openedx/core/djangoapps/embargo/middleware.py

```diff
@@ -83,15 +86,24 @@ def process_request(self, request):
             if pattern.match(request.path) is not None:
                 return None
 
-        ip_address = get_client_ip(request)[0]
+        if ip.USE_LEGACY_IP.is_enabled():
+            safest_ip_address = ip.get_legacy_ip(request)
+            all_ip_addresses = [safest_ip_address]
+        else:
+            safest_ip_address = ip.get_safest_client_ip(request)
+            all_ip_addresses = ip.get_all_client_ips(request)
+
         ip_filter = IPFilter.current()
 
-        if ip_filter.enabled and ip_address in ip_filter.blacklist_ips:
+        # When checking if a request is block-listed, we need to check EVERY client IP address in the chain, in case
+        # a blocked ip tried to hop through other proxies.
+        blocked_ips = [x for x in all_ip_addresses if x in ip_filter.blacklist_ips]
```

Comment:
So if a user in an allowed country proxies through an embargoed country, then they will be blocked? Do we want to only check the first and last IPs?

Reply:
> So if a user in an allowed country proxies through an embargoed country, then they will be blocked?

Correct.

> Do we want to only check the first and last IPs?

Checking first is "pointless" if you skip any, because any ip on the left of the chain can be spoofed. Checking only the last IP would be a reasonable thing to do, but then you let an embargoed user avoid the embargo with a single proxy.

Reply:
ok. I guess proxying through North Korea is a bad idea anyway